src/platform/datapath_raw_xdp.c

```diff
@@ -957,6 +1057,61 @@ CxPlatDpRawUninitialize(
     }
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+CxPlatDpRawOnSocketStateChange(
+    _In_ CXPLAT_SOCKET* Socket,
+    _In_ BOOLEAN IsCreated
+    )
+{
+    XDP_DATAPATH* Xdp = (XDP_DATAPATH*)Socket->Datapath;
+
+    if (Socket->Wildcard) {
+        const XDP_RULE Rule = {
+            .Match = XDP_MATCH_UDP_DST,
+            .Pattern.Port = Socket->LocalAddress.Ipv4.sin_port, // TODO - Check on on byte order
+            .Action = XDP_PROGRAM_ACTION_REDIRECT,
+            .Redirect.TargetType = XDP_REDIRECT_TARGET_TYPE_XSK,
+            .Redirect.Target = NULL,
+        };
+
+        CXPLAT_LIST_ENTRY* Entry;
+        for (Entry = Xdp->Interfaces.Flink; Entry != &Xdp->Interfaces; Entry = Entry->Flink) {
+            XDP_INTERFACE* Interface = CONTAINING_RECORD(Entry, XDP_INTERFACE, Link);
+            if (IsCreated) {
+                CxPlatDpRawInterfaceAddRule(Interface, &Rule);
```

Comment:
Is this guaranteed to succeed?

Reply:
It's best effort.

Reply:
But if this fails, won't that mean no connectivity? We ought to fail whatever QUIC API triggered this in that case.

Reply:
I just have to figure out clean up then, which might be really complicated. Let me look into it.

Reply:
@maolson-msft the problem comes down to `CxPlatDpRawInterfaceUpdateRules`. It applies the rules to all queues. What am I supposed to do if it fails on the 2nd+ queue (i.e. passed on at least one)?

Reply:
@maolson-msft & @csujedihy how about I just add a TODO to that function for now to figure out how to handle failure? Maybe we work with XDP to improve this so that we can update the rules for multiple queues as an atomic operation. That'd make this a lot easier.

Reply:
Sounds good to me